src/legacy/core_plugins/kbn_vislib_vis_types/public/types.ts

```diff
@@ -30,6 +30,13 @@ interface Labels {
   show: boolean;
   truncate: number;
 }
+interface ThresholdLine {
+  show: boolean;
+  value: number;
+  width: number;
+  style: string;
```

Comment:
Let's make this type `'full' | 'dashed' | 'dot-dashed'` instead, to be clear about the choices (also the editor will require this in the `SelectOptions` to work properly.